File: meituan-seckill-admin/src/api/tasks.ts
import { callApi } from './index'

export interface Task {
    id: number
    name: string
    post_url: string
    execution_time: string  // 格式: "HH:MM:SS:mmm"
    frequency: number
    requests_per_task: number
    priority: number
    is_active: boolean
    created_at: string
    updated_at: string | null
}

export interface TaskListResponse {
    items: Task[]
    total: number
    page: number
    size: number
}

export interface CreateTaskParams {
    name: string
    post_url: string
    execution_time: string  // 格式: "HH:MM:SS:mmm"
    frequency: number
    requests_per_task: number
    priority: number
    is_active: boolean
}

export interface UpdateTaskParams {
    name: string
    post_url: string
    execution_time: string
    frequency: number
    requests_per_task: number
    priority: number
    is_active: boolean
}

export interface UpdateTaskStatusParams {
    is_active: boolean
}

// 获取任务列表
export const getTaskList = async () => {
    console.log('调用getTaskList API');
    try {
        const response = await callApi('get', '/tasks');
        console.log('获取任务列表原始响应:', response);
        return response;
    } catch (error) {
        console.error('getTaskList API错误:', error);
        throw error;
    }
}

// 获取单个任务
export const getTask = async (id: number) => {
    return await callApi('get', `/tasks/${id}`);
}

// 创建任务
export const createTask = async (params: CreateTaskParams) => {
    console.log('创建任务参数:', params);
    try {
        const response = await callApi('post', '/tasks', params);
        console.log('创建任务响应:', response);
        return response;
    } catch (error) {
        console.error('创建任务错误:', error);
        throw error;
    }
}

// 更新任务
export const updateTask = async (id: number, params: UpdateTaskParams) => {
    console.log('更新任务参数:', id, params);
    try {
        const response = await callApi('put', `/tasks/${id}`, params);
        console.log('更新任务响应:', response);
        return response;
    } catch (error) {
        console.error('更新任务错误:', error);
        throw error;
    }
}

// 删除任务
export const deleteTask = async (id: number) => {
    console.log('删除任务ID:', id);
    try {
        const response = await callApi('delete', `/tasks/${id}`);
        console.log('删除任务响应:', response);
        return response;
    } catch (error) {
        console.error('删除任务错误:', error);
        throw error;
    }
}

// 修改任务状态
export const updateTaskStatus = async (id: number, params: UpdateTaskStatusParams) => {
    console.log('更新任务状态参数:', id, params);
    try {
        const response = await callApi('patch', `/tasks/${id}/status`, params);
        console.log('更新任务状态响应:', response);
        return response;
    } catch (error) {
        console.error('更新任务状态错误:', error);
        throw error;
    }
} 

File: meituan_seckill_api/app/routers/tasks.py
from typing import Any, List

from fastapi import APIRouter, Depends, HTTPException, Query, Request
from sqlalchemy.orm import Session

from app import crud, models, schemas
from app.core.deps import get_current_active_user, get_current_admin_user, get_api_key
from app.db.database import get_db

router = APIRouter()

@router.get("/", response_model=schemas.PaginatedResponse[schemas.Task])
def read_tasks(
    db: Session = Depends(get_db),
    skip: int = 0,
    limit: int = 100,
    current_user: models.User = Depends(get_current_active_user)
) -> Any:
    """
    获取任务列表
    """
    tasks = crud.task.get_multi(db, skip=skip, limit=limit)
    total = crud.task.get_count(db)
    return schemas.PaginatedResponse(
        items=tasks,
        total=total,
        page=skip // limit + 1 if limit > 0 else 1,
        size=limit
    )

@router.post("/", response_model=schemas.ApiResponse[schemas.Task])
def create_task(
    *,
    db: Session = Depends(get_db),
    task_in: schemas.TaskCreate,
    current_user: models.User = Depends(get_current_admin_user)
) -> Any:
    """
    创建新任务
    """
    task = crud.task.create(db=db, obj_in=task_in)
    return schemas.ApiResponse(data=task)

@router.get("/{id}", response_model=schemas.ApiResponse[schemas.Task])
def read_task(
    *,
    db: Session = Depends(get_db),
    id: int,
    current_user: models.User = Depends(get_current_active_user)
) -> Any:
    """
    获取特定任务
    """
    task = crud.task.get(db=db, id=id)
    if not task:
        raise HTTPException(status_code=404, detail="任务不存在")
    return schemas.ApiResponse(data=task)

@router.put("/{id}", response_model=schemas.ApiResponse[schemas.Task])
def update_task(
    *,
    db: Session = Depends(get_db),
    id: int,
    task_in: schemas.TaskUpdate,
    current_user: models.User = Depends(get_current_admin_user)
) -> Any:
    """
    更新任务
    """
    task = crud.task.get(db=db, id=id)
    if not task:
        raise HTTPException(status_code=404, detail="任务不存在")
    task = crud.task.update(db=db, db_obj=task, obj_in=task_in)
    return schemas.ApiResponse(data=task)

@router.delete("/{id}", response_model=schemas.ApiResponse)
def delete_task(
    *,
    db: Session = Depends(get_db),
    id: int,
    current_user: models.User = Depends(get_current_admin_user)
) -> Any:
    """
    删除任务
    """
    task = crud.task.get(db=db, id=id)
    if not task:
        raise HTTPException(status_code=404, detail="任务不存在")
    crud.task.remove(db=db, id=id)
    return schemas.ApiResponse(msg="删除成功")

@router.patch("/{id}/status", response_model=schemas.ApiResponse[schemas.Task])
def update_task_status(
    *,
    db: Session = Depends(get_db),
    id: int,
    status_in: schemas.TaskStatusUpdate,
    current_user: models.User = Depends(get_current_admin_user)
) -> Any:
    """
    更新任务状态（启用/禁用）
    """
    task = crud.task.get(db=db, id=id)
    if not task:
        raise HTTPException(status_code=404, detail="任务不存在")
    task = crud.task.update_task_status(db=db, task_id=id, is_active=status_in.is_active)
    return schemas.ApiResponse(data=task)

# API密钥访问的任务路由 
@router.get("/api/active", response_model=schemas.ApiResponse[List[schemas.Task]])
def read_active_tasks_api(
    db: Session = Depends(get_db),
    api_key: models.ApiKey = Depends(get_api_key),
    request: Request = None
) -> Any:
    """
    通过API密钥获取活跃任务
    """
    tasks = crud.task.get_active_tasks(db)
    return schemas.ApiResponse(data=tasks) 